scripts/west_commands/sign.py

```diff
@@ -213,7 +212,8 @@ def sign(self, command, build_dir, build_conf, formats):
         b = pathlib.Path(build_dir)
 
         imgtool = self.find_imgtool(command, args)
-        # The vector table offset is set in Kconfig:
+        # The vector table offset and application version are set in Kconfig:
+        appver = self.get_cfg(command, build_conf, 'CONFIG_APPLICATION_VERSION')
```

Comment:
Zephyr should not just accept to be restricted by the version format imposed by imgtool.

Zephyr uses various other tools which may support other version forms, such as CMake which uses `major.minor.patch.tweak` format.

Another popular variant is semantic versioning allows a `-<pre-release>` like `-rc1`.

Just like I don't want to restrict Zephyr's application version scheme to suit CMake, then we should not restrict it to suit `imgtool`.
Thus, the version format required by imgtool should not have this hard coupling to the version format offered by Zephyr.
Please use a imgtool specific setting here. That setting can then be a subset of application version whenever appllication version contains more fields that what is accepted by imgtool.

Reply:
As mentioned above, the input restriction for this is determined by the system being used, imgtool has a specific requirement for `a.b.c+d`, another bootloader might have a requirement for `a`, another might have a requirement for `a.b` - it makes no sense to have 2 completely different application versions, on in the application then one for a firmware update system whereby an end user is completely confused because they see multiple versions in a list e.g. 1.1.1.0, 1.1.2.0, 1.1.2.3, 1.1.2.5, 1.1.5.0 but then the device shows it is running on version 1.1 - what use is that?

Reply:
That doesn't explain anything, if a user application has a bootloader and e.g. bluetooth transport that can only show `a.b` - what use is the c.d in `a.b.c.d`? Each thing having its own Kconfig is a complete and utter mess.